src/storage/store/string_column_chunk.cpp

```diff
@@ -28,7 +52,8 @@ void StringChunkData::append(ValueVector* vector, const SelectionVector& selVect
         KU_ASSERT(vector->dataType.getPhysicalType() == PhysicalTypeID::STRING);
         // index is stored in main chunk, data is stored in the data chunk
         nullChunk->setNull(numValues, vector->isNull(pos));
```

Comment:
I kept the null chunk in the main column populated (and the index column's null chunk is unpopulated). Is this ok or should the null chunk also be moved to the index column?

Reply:
That sounds fine to me. One side-effect of not storing the nulls alongside an integer column is that null values will be compressed explicitly, however for the index data it should always be in the range of `[0, dictionarySize)`, so it shouldn't make a significant difference to compression efficiency if the presence of null values prevents us from using frame of reference compression by forcing a minimum value of 0.

You can also omit the index column's null column entirely if you use the Column constructor directly (like what's done in DictionaryColumn) and do similarly for the MetadataDAHInfo in [table_statistics_collection.cpp](https://github.com/kuzudb/kuzu/pull/3617/files#diff-f18bee90bf3f57e00224cb9d0287d548c9fa9aa980b4e4ac473a4e27f7a72253). The factory is only needed if the type isn't hardcoded.

Reply:
Cool, I'll keep the null column in the main column and remove the null column from the index column then

Reply:
updated